### PR TITLE
Docs Improvements - Add azure repos

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
---
* Added Azure Repos section

### CODE REVIEW

File: docs/install-azure-repos.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
+# Setting up Frogbot on Azure Repos
```
[the remark this answers is not in the document]
Comment (omerzi) [reply docs/install-azure-repos.md:1]: done

File: docs/install-azure-repos.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+5. Select the relevant repository of which Frogbot will scan.
+6. Select 'Starter Pipeline' and use `frogbot-scan-pr.yml` and `frogbot-scan-and-fix.yml`:
+
+#### frogbot-scan-pr.yml:
+
+```yaml
+trigger:
+  branches:
+    include:
+      - master
```
[the remark this answers is not in the document]
Comment (omerzi) [reply docs/install-azure-repos.md:23]: done

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: docs/install-azure-repos.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
+[Go back to the main documentation page](../README.md)
+# Setting up Frogbot on Azure Repos
```
[the remark this answers is not in the document]
Comment (omerzi) [reply docs/install-azure-repos.md]: done

File: docs/install-azure-repos.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+To install Frogbot on Azure Repos repositories:
+
+1. Make sure you have the connection details of your JFrog environment.
+2. Go to your Azure DevOps repository settings page and save the JFrog connection details as repository secrets with the
+   following names - JF_URL, JF_USER, and JF_PASSWORD
+
+> **_NOTE:_** You can also use **JF_XRAY_URL** and **JF_ARTIFACTORY_URL** instead of **JF_URL**, and **JF_ACCESS_TOKEN**
+> instead of **JF_USER** and **JF_PASSWORD**
```
[the remark this answers is not in the document]
Comment (omerzi) [reply docs/install-azure-repos.md]: done

File: docs/install-azure-repos.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+3. Go to Azure DevOps Pipelines of the relevant project, and add `New pipeline`.
+4. Set `Azure Repos Git` as your code source.
+5. Select the relevant repository of which Frogbot will scan.
```
[the remark this answers is not in the document]
Comment (omerzi) [reply docs/install-azure-repos.md]: done

File: docs/install-azure-repos.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
+[Go back to the main documentation page](../README.md)
+# Installing Frogbot on Azure Repos repositories
+
+To install Frogbot on Azure Repos repositories:
+
+1. Go to Azure DevOps Pipelines of the relevant project, and add `New pipeline`.
```
Comment (eyalbe4) [inline docs/install-azure-repos.md:6]: Go to Azure DevOps Pipelines of the relevant project, and add `New pipeline`.
-->
Go to your Azure Pipelines project, and add a new pipeline.

Let's also add the relevant image from the Azire DevOps UI.

File: docs/install-azure-repos.md [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+2. Set `Azure Repos Git` as your code source.
+3. Select the repository you'd like Frogbot to scan.
+4. Select 'Starter Pipeline' and use `frogbot-scan-pr.yml` and `frogbot-scan-and-fix.yml`:
```
Comment (eyalbe4) [inline docs/install-azure-repos.md:9]: Let's replace the above section with the following two sections:

4. Select 'Starter Pipeline' and name it 'frogbot-scan-pr'. Use the content of the below yaml for the pipeline.
[expandable yml content]

5. Select 'Starter Pipeline' and name it 'frogbot-scan-and-fix'. Use the content of the below yaml for the pipeline.
[expandable yml content]

File: docs/install-azure-repos.md [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+5. Make sure you have the connection details of your JFrog environment.
```
Comment (eyalbe4) [inline docs/install-azure-repos.md:10]: I think this should be the very first step.

File: docs/install-azure-repos.md [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+6. In the pipeline page save the JFrog connection details as variables with the
+   following names - JF_URL, JF_USER, and JF_PASSWORD.
+
+> **_NOTE:_** You can also use **JF_XRAY_URL** and **JF_ARTIFACTORY_URL** instead of **JF_URL**, and **JF_ACCESS_TOKEN**
+> instead of **JF_USER** and **JF_PASSWORD**.
+
+7. To set variables in the pipeline page, click on `Variables` button and set `New variable`:
```
Comment (eyalbe4) [inline docs/install-azure-repos.md:17]: It looks like section and sections 7 above should be mrged somehow, and the images should help the user understand when to find and set the variables.

File: docs/install-azure-repos.md [hunk 10]
```diff
@@ -0,0 +1,180 @@
+[Go back to the main documentation page](../README.md)
+# Installing Frogbot on Azure Repos repositories
+
+To install Frogbot on Azure Repos repositories:
+
+1. Go to Azure DevOps Pipelines of the relevant project, and add `New pipeline`.
+2. Set `Azure Repos Git` as your code source.
```
Comment (eyalbe4) [inline docs/install-azure-repos.md:7]: Let's add the relevant image from the Azire DevOps UI.

File: docs/install-azure-repos.md [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+3. Select the repository you'd like Frogbot to scan.
```
Comment (eyalbe4) [inline docs/install-azure-repos.md:8]: Let's add the relevant image from the Azire DevOps UI.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: docs/install-azure-repos.md (hunk 6 above)
Comment (omerzi) [reply docs/install-azure-repos.md:6]: done

File: docs/install-azure-repos.md (hunk 7 above)
Comment (omerzi) [reply docs/install-azure-repos.md:9]: done

File: docs/install-azure-repos.md (hunk 8 above)
Comment (omerzi) [reply docs/install-azure-repos.md:10]: done

File: docs/install-azure-repos.md (hunk 9 above)
Comment (omerzi) [reply docs/install-azure-repos.md:17]: done

File: docs/install-azure-repos.md (hunk 11 above)
Comment (omerzi) [reply docs/install-azure-repos.md]: done

File: docs/install-azure-repos.md [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+
+   ![azure-set-code-source.png.png](../images/azure-set-code-source.png)
+
+
+3. Select the repository you'd like Frogbot to scan.
+
+   ![azure-select-repo-to-test.png](../images/azure-select-repo-to-test.png)
+
+
+4. Select `Starter Pipeline` and name it `frogbot-scan-pr.yml`. Use the content of the below yaml for the pipeline.
```
Comment (eyalbe4) [inline docs/install-azure-repos.md:22]: frogbot-scan-pr.yml --> frogbot-scan-pr
(I assume we wouldn't want to have the pipeline name onclude the .yml, right?)

The same goes for section 5 below.

Comment (eyalbe4) [inline docs/install-azure-repos.md:22]: ... the below yaml for the pipeline. --> ... the below template for the pipeline.
The same goes for section 5 below.

File: docs/install-azure-repos.md [hunk 13]
```diff
@@ -0,0 +1,202 @@
...
+
+          inputs:
+             script: |
+                curl -fLg "https://releases.jfrog.io/artifactory/frogbot/v2/[RELEASE]/getFrogbot.sh" | sh
+                ./frogbot cfpr
+```
+
+</details>
+
+6. Make sure you have the connection details of your JFrog environment. In the pipeline page save the JFrog connection details as variables with the following names - JF_URL, JF_USER, and JF_PASSWORD.
```
Comment (eyalbe4) [inline docs/install-azure-repos.md:189]: I think this section should be split as follows:

We should have the followinf as step 1 in the instructions -
Make sure you have the connection details of your JFrog environment

And the following should be right after section 3 - 

"
In the Pipeline page, save the JFrog connection details as variables with the following names - JF_URL, JF_USER, and JF_PASSWORD.
"

File: docs/install-azure-repos.md [hunk 14]
```diff
@@ -0,0 +1,202 @@
...
+   To set variables in the pipeline edit page, click on `Variables` button:
+
+   ![variables_button.png](../images/azure-variables-button.png)
+
+   Set `New variable`:
+
+   ![img_1.png](../images/azure-new-variable.png)
+
+> **_NOTE:_** You can also use **JF_XRAY_URL** and **JF_ARTIFACTORY_URL** instead of **JF_URL**, and **JF_ACCESS_TOKEN**
+> instead of **JF_USER** and **JF_PASSWORD**.
```
Comment (eyalbe4) [inline docs/install-azure-repos.md:200]: This "Note" comment probably should probably be before the images, and right after 
"
Make sure you have the connection details of your JFrog environment. In the pipeline page save the JFrog connection details as variables with the following names - JF_URL, JF_USER, and JF_PASSWORD.
"

File: docs/install-azure-repos.md [hunk 15]
```diff
@@ -0,0 +1,202 @@
...
+
+7. Set the other mandatory `Variables` according to the chosen pipeline template.
```
Comment (eyalbe4) [inline docs/install-azure-repos.md:202]: Set the other mandatory `Variables` according to the chosen pipeline template.
-->
Edit the yaml of the pipeline you created, and set the relevant branches to be scanned, as well as the remaining mandatory variables.

Let's also duplicate this section for each of the two pipelines, and place it right after the section that adds the template yaml to the pipeline.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: docs/install-azure-repos.md (hunk 12 above)
Comment (omerzi) [reply docs/install-azure-repos.md:22]: done

Comment (omerzi) [reply docs/install-azure-repos.md:22]: done

File: docs/install-azure-repos.md (hunk 14 above)
Comment (omerzi) [reply docs/install-azure-repos.md:200]: done

File: docs/install-azure-repos.md (hunk 15 above)
Comment (omerzi) [reply docs/install-azure-repos.md:202]: done

File: docs/install-azure-repos.md [hunk 16]
```diff
@@ -0,0 +1,230 @@
+[Go back to the main documentation page](../README.md)
+
+# Installing Frogbot on Azure Repos repositories
+
+To install Frogbot on Azure Repos repositories:
```
Comment (eyalbe4) [inline docs/install-azure-repos.md:5]: To install Frogbot on Azure Repos repositories:
-->
To install Frogbot on Azure Repos repositories, follow these steps.

File: docs/install-azure-repos.md [hunk 17]
```diff
@@ -0,0 +1,230 @@
...
+
+**Step 1:** Make sure you have the connection details of your JFrog environment.
```
Comment (eyalbe4) [inline docs/install-azure-repos.md:7]: Let's add a new-line between the step title and the step test and also replace the "bold" wrapping around the step number with a "title" markdown annotation - ```## Step 1``` . The above therefore should be changed to - 
## Step 1
Make sure you have the connection details of your JFrog environment.

File: docs/install-azure-repos.md [hunk 18]
```diff
@@ -0,0 +1,230 @@
...
+
+To set the `Variables` in the pipeline edit page, click on the `Variables` button and set the `Variables`.
+
+![variables_button.png](../images/azure-variables-button.png)
+
+![img_1.png](../images/azure-new-variable.png)
+
+**Step 11:** To enable pull request scanning, you must set up `Branch Policies` for the relevant target branches Frogbot should scan.
+
+**Step 12:** Go to Azure Repos -> Branches.
```
Comment (eyalbe4) [inline docs/install-azure-repos.md:214]: Step 11 and Step 12 should be merged into one step, beucase Step 111 currently includes no action.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: docs/install-azure-repos.md (hunk 16 above)
Comment (omerzi) [reply docs/install-azure-repos.md:5]: done

File: docs/install-azure-repos.md (hunk 17 above)
Comment (omerzi) [reply docs/install-azure-repos.md:7]: done

File: docs/install-azure-repos.md (hunk 18 above)
Comment (omerzi) [reply docs/install-azure-repos.md:214]: done

**Pull request merged**